### PR TITLE
Ensure data parsed as JSON is a dict

### PR DESCRIPTION
**Pull request by emontnemery**

### CODE REVIEW

File: pychromecast/socket_client.py [hunk 1]
```diff
@@ -79,10 +79,13 @@ class InterruptLoop(Exception):
...
 
-def _json_from_message(message):
-    """ Parses a PB2 message into JSON format. """
+def _dict_from_message(message):
+    """ Parses a PB2 message as a JSON dict. """
     try:
-        return json.loads(message.payload_utf8)
+        dict_data = json.loads(message.payload_utf8)
+        if not isinstance(dict_data, dict):
+            return {}
```
Comment (balloob) [inline pychromecast/socket_client.py:87]: Should we log a warning or at least debug?

Comment (emontnemery) [reply pychromecast/socket_client.py:93]: Fixed!

File: pychromecast/socket_client.py [hunk 2]
```diff
@@ -79,14 +79,23 @@ class InterruptLoop(Exception):
...
-def _json_from_message(message):
-    """ Parses a PB2 message into JSON format. """
+def _dict_from_message_payload(message):
+    """ Parses a PB2 message as a JSON dict. """
     try:
-        return json.loads(message.payload_utf8)
+        data = json.loads(message.payload_utf8)
+        if not isinstance(data, dict):
+            logger = logging.getLogger(__name__)
+            logger.info(
```
Comment (balloob) [inline pychromecast/socket_client.py:88]: Should we make this a warning?

Comment (balloob) [reply pychromecast/socket_client.py:88]: Or debug?

**Pull request merged**